datasets/gb/nca_enforcements/crawler.py

```diff
@@ -0,0 +1,137 @@
+- entity_schema: {schema_field.description}
+- address: The locations associated with the offender, such as their home address, town, city, or region."
+- country: Any countries the entity is indicated to reside, operate, or have been born or registered in. Leave empty if not explicitly stated.
+"""
+
+
+def crawl_enforcement_action(context: Context, url: str) -> None:
+    article_doc = context.fetch_html(url, cache_days=7, absolute_links=True)
+    if article_doc is None:
+        return
+    article_name = article_doc.xpath(NAME_XPATH)[0]
+    assert article_name, "Article name not found"
+    article_content = article_doc.xpath(CONTENT_XPATH)
+    assert len(article_content) == 1
+    article_element = article_content[0]
+    date = article_doc.xpath(DATE_XPATH)
+    # The date is absent in some articles
+    if date != []:
+        date = date[0]
+    # Extract topics and look them up
+    raw_topics: List[str] = article_doc.xpath(TOPIC_XPATH)
+    topics: List[str] = []
+    for topic in raw_topics:
+        res = context.lookup("topics", topic)
+        if res is not None:
+            topics.append(res.value)
+        else:
+            context.log.warning("Topic lookup not found", topic=topic)
+
+    source_value = HtmlSourceValue(
+        key_parts=url,
+        label="Press Release",
+        element=article_element,
+        url=url,
+    )
+    prompt_result = run_typed_text_prompt(
+        context, PROMPT, source_value.value_string, Offenders
+    )
+    review = review_extraction(
+        context,
+        source_value=source_value,
+        original_extraction=prompt_result,
+        origin=DEFAULT_MODEL,
+    )
+
+    if not review.accepted:
+        return
+
+    for item in review.extracted_data.offenders:
+        entity = context.make(item.entity_schema)
+        entity.id = context.make_id(item.name, str(item.address), str(item.country))
+        entity.add("name", item.name, origin=review.origin)
+        if item.address != item.country:
+            entity.add("address", item.address, origin=review.origin)
+        entity.add("country", item.country, origin=review.origin)
+        for topic in topics:
+            entity.add("topics", topic)
+
+        # We use the date as a key to make sure notices about separate actions are separate sanction entities
+        sanction = h.make_sanction(context, entity, date)
+        h.apply_date(sanction, "date", date)
+        sanction.set("sourceUrl", url)
+
+        article = h.make_article(context, url, title=article_name, published_at=date)
+        documentation = h.make_documentation(context, entity, article)
+
+        context.emit(entity)
+        context.emit(sanction)
+        context.emit(article)
+        context.emit(documentation)
+
+
+def crawl(context: Context):
+    page: int = 0
+    base_url: str = "https://www.nationalcrimeagency.gov.uk/news?start={offset}"
+
+    while True:
+        index_url = base_url.format(offset=page * 16)
```

Comment:
What if the default number of items on the page changes? Does it make sense to count `links` instead?